### PR TITLE
import-chrome: import zone metadata from "args" key

### PR DESCRIPTION
**Pull request by ethercrow**
Fixes https://github.com/wolfpld/tracy/issues/20

Screen Shot 2020-05-12 at 3 47 15 PM

### CODE REVIEW

**Comment by wolfpld** [comment]
`std::move` should be used for `zoneText`.

File: import-chrome/src/import-chrome.cpp [hunk 1]
```diff
@@ -70,13 +70,23 @@ int main( int argc, char** argv )
     for( auto& v : j )
     {
         const auto type = v["ph"].get<std::string>();
+
+        std::string zoneText = "";
+        if ( v.contains( "args" ) )
+        {
+            for ( auto& kv : v["args"].items() )
+            {
+              zoneText += kv.key() + ": " + kv.value().dump() + "\n";
```
Comment (wolfpld) [inline import-chrome/src/import-chrome.cpp:79]: Please fix indentation.

**Comment by ethercrow** [comment]
I'm rather casual C++ user, updated.

**Comment by ethercrow** [comment]
Is it fine to use repeated `+=` for `std::string` concatenation?

File: import-chrome/src/import-chrome.cpp [hunk 2]
```diff
@@ -96,7 +106,7 @@ int main( int argc, char** argv )
             const auto ts0 = uint64_t( v["ts"].get<double>() * 1000. );
             const auto ts1 = v["dur"].is_object() ? ts0 + uint64_t( v["dur"].get<double>() * 1000. ) : ts0;
             const auto name = v["name"].get<std::string>();
-            timeline.emplace_back( tracy::Worker::ImportEventTimeline { tid, ts0, name, "", false } );
+            timeline.emplace_back( tracy::Worker::ImportEventTimeline { tid, ts0, name, zoneText, false } );
```
Comment (wolfpld) [inline import-chrome/src/import-chrome.cpp:109]: Also here :)

Comment (ethercrow) [reply import-chrome/src/import-chrome.cpp]: 👍

**Comment by wolfpld** [comment]
In general case it is not recommended, as it involves memory allocation, but in this case it won't matter that much. Chrome traces are tiny.

**Pull request merged**